changelog.d/21201_fields_iter.fix.md

```diff
@@ -0,0 +1 @@
+The `reduce` transform can now reduce fields that contain special characters.
```

Comment:
FYI @jszwedko I accepted your suggestion but then I realized this never paniced, it just failed to reduce when the path was not quoted.